src/java/org/apache/cassandra/streaming/async/StreamingMultiplexedChannel.java

```diff
@@ -202,7 +210,9 @@ public Future<?> sendMessage(StreamingChannel channel, StreamMessage message)
                 throw new RuntimeException("Cannot send stream data messages for preview streaming sessions");
             if (logger.isDebugEnabled())
                 logger.debug("{} Sending {}", createLogTag(session), message);
-            return fileTransferExecutor.submit(new FileStreamTask((OutgoingStreamMessage)message));
+
+            InetAddressAndPort connectTo = SystemKeyspace.getPreferredIP(to);
```

Comment:
do we know If there are any paths where this is called on a Netty thread?

Reply:
`StreamSession#startStreamingFiles()` is the only thing that can call `sendMessage()` here with a `OutgoingStreamMessage`. I don't think we can hit that from an even loop thread, but I'll do a quick audit...